x/mongo/driver/topology/pool_test.go

```diff
@@ -816,6 +816,79 @@ func TestPool(t *testing.T) {
 				assert.Containsf(t, err.Error(), "canceled", `expected error message to contain "canceled"`)
 			}
 
+			p.close(context.Background())
+		})
+		t.Run("discards connections closed by the server side", func(t *testing.T) {
+			t.Parallel()
+
+			cleanup := make(chan struct{})
+			defer close(cleanup)
+
+			ncs := make(chan net.Conn, 2)
+			addr := bootstrapConnections(t, 2, func(nc net.Conn) {
+				// Send all "server-side" connections to a channel so we can
+				// interact with them during the test.
+				ncs <- nc
+
+				<-cleanup
+				_ = nc.Close()
+			})
+
+			d := newdialer(&net.Dialer{})
+			p := newPool(poolConfig{
+				Address: address.Address(addr.String()),
+			}, WithDialer(func(Dialer) Dialer { return d }))
+			err := p.ready()
+			noerr(t, err)
```

Comment:
Suggest using the `require` package instead of `noerr`.

Reply:
Replaced all uses of `noerr` with `require.NoError`